### PR TITLE
PSC-5 Created protocol and example of router

### PR DESCRIPTION
**Pull request by Pryssss**

### CODE REVIEW

File: Uni/Business Logic/Router/Unirouter.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+//  Unirouter.swift
+//  Uni
+//
+//  Created by Markiyan Prysiazhniuk on 09.10.2019.
+//
+
+import Foundation
+
+enum UniRouter: RouterProtocol {
+    private enum Constants {
```
Comment (Alex-Chaku) [inline Uni/Business Logic/Router/Unirouter.swift]: Add full demo of Uni as struct

File: Uni/Business Logic/Router/Protocols/RouterProtocol.swift [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    var postBody: Encodable? {
+        return nil
+    }
+}
+
+extension RouterProtocol {
+    func asURLRequest() throws -> URLRequest {
+        guard var componentsURL = URLComponents(string: baseURL + path) else { throw RouterError.failedToCreateURL }
+
+        if let queryItems = queryItems {
```
Comment (Pryssss) [inline Uni/Business Logic/Router/Protocols/RouterProtocol.swift]: тут спростити

File: Uni/Business Logic/Router/Protocols/RouterProtocol.swift [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+//  Uni
+//
+//  Created by Markiyan Prysiazhniuk on 09.10.2019.
+//
+
+import Foundation
+
+// MARK: - Header naming
+public typealias HTTPHeaders = [String: String]
+public typealias QureyItems = [String: String]?
```
Comment (Alex-Chaku) [inline Uni/Business Logic/Router/Protocols/RouterProtocol.swift]: Why it's optional when default value is an empty dictionary. In general, it's better to use `typealias` only for types, without optionality

File: Uni/Business Logic/Router/Protocols/RouterProtocol.swift [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+// MARK: - as URLRequest extention
+
+extension RouterProtocol {
+
+    func asURLRequest() throws -> URLRequest {
+        guard var componentsURL = URLComponents(string: baseURL + path) else { throw RouterError.failedToCreateURL }
+
+        if let queryItems = queryItems {
+            let queryItems = queryItems.map { URLQueryItem(name: $0.key, value: $0.value) }
```
Comment (Alex-Chaku) [inline Uni/Business Logic/Router/Protocols/RouterProtocol.swift]: You can do just 1 line, without excessive let

File: Uni/Business Logic/Router/Protocols/RouterProtocol.swift [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+// MARK: - Header naming
+public typealias HTTPHeaders = [String: String]
+public typealias QureyItems = [String: String]?
+
+// MARK: - RouterError
+enum RouterError: Error {
+
+    case failedToCreateURL
+    case invalidURL(url: String)
```
Comment (Alex-Chaku) [inline Uni/Business Logic/Router/Protocols/RouterProtocol.swift]: Add error with bad post body

File: Uni/Business Logic/Router/Protocols/RouterProtocol.swift [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+            componentsURL.queryItems = queryItems
+        }
+        guard let requestURL = URL(string: componentsURL.description) else { throw RouterError.invalidURL(url: componentsURL.description) }
+
+        var request = URLRequest(url: requestURL)
+        request.httpMethod = method.rawValue
+        request.allHTTPHeaderFields = headers
+
+        if let postBody = postBody {
+            let postData = try postBody.encoded()
```
Comment (Alex-Chaku) [inline Uni/Business Logic/Router/Protocols/RouterProtocol.swift]: Add here guard + throw if post body, not nil and can't be encoded

**Pull request merged**